Split/Storage/CoreDataHelper.swift

```diff
@@ -72,12 +72,21 @@ class CoreDataHelper {
         if let predicate = predicate {
             fetchRequest.predicate = predicate
         }
-        let deleteRequest = NSBatchDeleteRequest(fetchRequest: fetchRequest)
-
         do {
-            try persistentCoordinator.execute(deleteRequest, with: managedObjectContext)
+            let entities = try managedObjectContext.fetch(fetchRequest).compactMap { return $0 as? NSManagedObject }
+            for entity in entities {
+                managedObjectContext.delete(entity)
+            }
         } catch {
             Logger.e("Error while deleting \(entity.rawValue) entities from storage: \(error.localizedDescription)")
         }
+
+        //        let deleteRequest = NSBatchDeleteRequest(fetchRequest: fetchRequest)
```

Comment:
this comment could be removed if we are not keeping it for future references

Reply:
Done!